repo/blob/azure/azure_pit.go

```diff
@@ -181,21 +184,17 @@ func maybePointInTimeStore(ctx context.Context, s *azStorage, pointInTime *time.
 		return s, nil
 	}
 
-	// Versioning is needed for PIT. This check will fail if someone deleted the Kopia Repository file.
-	props, err := s.service.ServiceClient().
-		NewContainerClient(s.container).
-		NewBlobClient(s.getObjectNameString(format.KopiaRepositoryBlobID)).
-		GetProperties(ctx, nil)
-	if err != nil {
-		return nil, errors.Wrapf(err, "could not get determine if container '%s' supports versioning", s.container)
+	pit := &azPointInTimeStorage{
+		azStorage:   *s,
+		pointInTime: *pointInTime, // not used for the check
 	}
 
-	if props.VersionID == nil {
-		return nil, errors.Errorf("cannot create point-in-time view for non-versioned container '%s'", s.container)
+	err := pit.getBlobVersions(ctx, format.KopiaRepositoryBlobID, func(vm versionMetadata) error {
```

Comment:
Mike: how does these behave in the following scenarios:

- the container does not have versioning enabled?
- the container has versioning enabled?

Please add unit tests for these scenarios, for `maybePointInTimeStore` and `az.getVersionedBlobMeta`, to run them as part of the "core provider tests".

Thanks.

Reply:
the container does not have versioning enabled? => PIT fails. Added a test case

the container has versioning enabled? => it works. the existing test cases cover this although I modified it to delete the kopia.repository file to ensure it still works when the file is deleted